bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -571,7 +571,11 @@ public void safeRun() {
 
                 // error out all pending adds during closing, the callbacks shouldn't be
                 // running under any bk locks.
-                errorOutPendingAdds(rc, pendingAdds);
+                try {
+                    errorOutPendingAdds(rc, pendingAdds);
+                } catch (Exception e) {
+                    closePromise.completeExceptionally(e);
```

Comment:
If there has an exception, do we need to do the `tearDownWriteHandleState()` and ` MetadataUpdateLoop` operations?

Reply:
Yes, it still needs the following operations. Catch this exception just to ensure the closePromise can complete.

Reply:
we shouldn't "completeExceptionally" or "complete" the `closePromise` twice